File: android/src/main/java/com/taihua/hud/HUDConfig.java
package com.taihua.hud;

import android.graphics.Color;
import androidx.annotation.ColorInt;

public class HUDConfig {

    static int DEFAULT_DURATION = 2000;
    static int DEFAULT_GRACE_TIME = 300;
    static int DEFAULT_MIN_SHOW_TIME = 800;
    static int DEFAULT_BACKGROUND_COLOR = Color.parseColor("#b1000000");
    static float DEFAULT_CORNER_RADIUS = 5;
    static @ColorInt
    int DEFAULT_TINT_COLOR = Color.WHITE;
    static String DEFAULT_LOADING_TEXT = null;

    public static int duration = DEFAULT_DURATION;
    public static int graceTime = DEFAULT_GRACE_TIME;
    public static int minShowTime = DEFAULT_MIN_SHOW_TIME;

    public static int backgroundColor = DEFAULT_BACKGROUND_COLOR;
    public static @ColorInt
    int tintColor = DEFAULT_TINT_COLOR;
    public static float cornerRadius = DEFAULT_CORNER_RADIUS;
    public static String loadingText = null;

}

File: ios/HudHybrid/HudConfig.m
//
//  HudConfig.m
//  HudHybrid
//
//  Created by 李生 on 2019/11/21.
//

#import "HudConfig.h"

@implementation HudConfig

+ (instancetype)sharedConfig {
    static HudConfig *config;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        config = [[HudConfig alloc] init];
    });
    return config;
}

+ (NSTimeInterval)defaultDuration {
    return 2.0f;
}

+ (NSTimeInterval)defaultGraceTime {
    return 0.3f;
}

+ (NSTimeInterval)defaultMinShowTime {
    return 0.8f;
}

+ (UIColor *)defaultBezelColor {
    return nil;
}

+ (UIColor *)defaultContentColor {
    return nil;
}

+ (CGFloat)defaultCornerRadius {
    return 10.f;
}

+ (NSString *)defaultLoadingText {
    return nil;
}

- (instancetype)init {
    if (self = [super init]) {
        _duration = [HudConfig defaultDuration];
        _graceTime = [HudConfig defaultGraceTime];
        _minshowTime = [HudConfig defaultMinShowTime];
        _bezelColor = [HudConfig defaultBezelColor];
        _contentColor = [HudConfig defaultContentColor];
        _cornerRadius = [HudConfig defaultCornerRadius];
        _loadingText = [HudConfig defaultLoadingText];
    }
    return self;
}

@end
